compiler/rustc_resolve/src/access_levels.rs

```diff
@@ -27,6 +28,11 @@ impl<'r, 'a> AccessLevelsVisitor<'r, 'a> {
         let mut visitor = AccessLevelsVisitor { r, changed: false };
 
         visitor.set_access_level_def_id(CRATE_DEF_ID, Some(AccessLevel::Public));
+        
+        let mut crate_effective_vis = EffectiveVisibilityValue::new();
+        crate_effective_vis.set(CRATE_DEF_ID, AccessLevel::Public);
+        visitor.r.effective_vis.set_effective_vis(CRATE_DEF_ID, crate_effective_vis);
```

Comment:
If something is public up to `def_id`, then it's exported/reachable at least up to `def_id` too, but these methods only set the `Public` field.
I think we should always keep the values in `EffectiveVisibilityValue` in self-consistent state and provide methods that update them simultaneously.

Reply:
Also, "public up to `CRATE_DEF_ID`" means `pub(crate)`, not `pub`.
So just a `LocalDefId` is not enough for an (effective) visibility, you need a fully unrestricted `Pub` variant too.
You can actually reuse `ty::Visibility` for this stuff, it already contains both a restricted variant with def-id and an unrestricted variant.

Reply:
`None` can be used as an optimization, but it's basically `Visibility::Restricted` with the closest module.

Reply:
`LocalDefId` in these tables should always point to a `mod` item (or crate root), not to non-module items themselves, that would be consistent with how visibilities are treated everywhere else.